tests/bluetooth/audio/ascs/src/test_ase_state_transition.c

```diff
@@ -365,17 +608,692 @@ ZTEST_F(test_sink_ase_state_transition, test_client_streaming_to_qos_configured)
+
+	/* Preamble */
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+	test_ase_control_client_config_qos(conn, ase_id);
+	test_mocks_reset();
+
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+
+	/* Verification */
+	expect_bt_bap_unicast_server_cb_reconfig_called_once(stream, BT_AUDIO_DIR_SOURCE, EMPTY);
+	expect_bt_bap_unicast_server_cb_config_not_called();
+	expect_bt_bap_unicast_server_cb_qos_not_called();
+	expect_bt_bap_unicast_server_cb_enable_not_called();
+	expect_bt_bap_unicast_server_cb_start_not_called();
+	expect_bt_bap_unicast_server_cb_metadata_not_called();
+	expect_bt_bap_unicast_server_cb_disable_not_called();
+	expect_bt_bap_unicast_server_cb_stop_not_called();
+	expect_bt_bap_unicast_server_cb_release_not_called();
+
+	expect_bt_bap_stream_ops_configured_called_once(stream, EMPTY);
+	expect_bt_bap_stream_ops_qos_set_not_called();
+	expect_bt_bap_stream_ops_enabled_not_called();
+	expect_bt_bap_stream_ops_metadata_updated_not_called();
+	expect_bt_bap_stream_ops_disabled_not_called();
+	expect_bt_bap_stream_ops_released_not_called();
+	expect_bt_bap_stream_ops_started_not_called();
+	expect_bt_bap_stream_ops_stopped_not_called();
+}
+
+ZTEST_F(test_source_ase_state_transition, test_client_codec_configured_to_releasing)
+{
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	uint8_t ase_id = fixture->ase.id;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SRC);
+
+	/* Preamble */
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+	test_mocks_reset();
+
+	test_ase_control_client_release(conn, ase_id);
+
+	/* Verification */
+	expect_bt_bap_unicast_server_cb_release_called_once(stream);
+	expect_bt_bap_unicast_server_cb_config_not_called();
+	expect_bt_bap_unicast_server_cb_reconfig_not_called();
+	expect_bt_bap_unicast_server_cb_qos_not_called();
+	expect_bt_bap_unicast_server_cb_enable_not_called();
+	expect_bt_bap_unicast_server_cb_start_not_called();
+	expect_bt_bap_unicast_server_cb_metadata_not_called();
+	expect_bt_bap_unicast_server_cb_disable_not_called();
+	expect_bt_bap_unicast_server_cb_stop_not_called();
+
+	expect_bt_bap_stream_ops_released_called_once(stream);
+	expect_bt_bap_stream_ops_configured_not_called();
+	expect_bt_bap_stream_ops_qos_set_not_called();
+	expect_bt_bap_stream_ops_enabled_not_called();
+	expect_bt_bap_stream_ops_metadata_updated_not_called();
+	expect_bt_bap_stream_ops_disabled_not_called();
+	expect_bt_bap_stream_ops_started_not_called();
+	expect_bt_bap_stream_ops_stopped_not_called();
+}
+
+ZTEST_F(test_source_ase_state_transition, test_client_enabling_to_releasing)
+{
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	uint8_t ase_id = fixture->ase.id;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SRC);
+
+	/* Preamble */
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+	test_ase_control_client_config_qos(conn, ase_id);
+	test_ase_control_client_enable(conn, ase_id);
```

Comment:
Would it make sense to define a helper function which calls these functions based on the wanted state? So that we could just do
```
test_ase_control_client_set_preamble_state(BT_BAP_EP_STATE_ENABLING);
```

Reply:
Because basically what this test really wants to do is
```
setup_preamble;
test_ase_control_client_release(conn, ase_id);

expect_bt_bap_unicast_server_cb_release_called_once(stream);
expect_bt_bap_stream_ops_released_called_once(stream);
```

but instead of being 4 (or maybe a few more) lines, it's 34. It's not necessarily bad to be verbose in testing, but I'm concerned about the number of lines that are essentially copied for each new test case, which could cause developers to forget one or 2.